File: stageTemplates/src/app/models/site.ts
import { Agence } from './agence'; // Make sure to provide the correct import path
import { TypeGrandeur } from './typegrandeur';

export class Site {
     id: number;
     nom: string;
     localisationBarr: string;
     capacite: number | null;
     type: string;
     villePlusProche: string;
     hauteurBarr: number | null;
     distVillePlusProche: number | null;
     codeRetNormal: string;
     dateMiseEnServ: Date | null; 
     laRetenue: string; 
     agenceId: number | null;
     agence?: Agence ;
    constructor(
         id: number,
         nom: string,
         localisationBarr: string,
         capacite: number | null,
         type: string, 
         villePlusProche: string,
         hauteurBarr: number | null,
         distVillePlusProche: number | null,
         codeRetNormal: string,
         dateMiseEnServ: Date | null, 
         laRetenue: string, 
         agenceId: number | null 
    ) {
        this.id = id;
        this.nom = nom;
        this.localisationBarr = localisationBarr;
        this.capacite = capacite;
        this.type = type;
        this.villePlusProche = villePlusProche;
        this.hauteurBarr = hauteurBarr;
        this.distVillePlusProche = distVillePlusProche;
        this.codeRetNormal=codeRetNormal;
        this.dateMiseEnServ=dateMiseEnServ;
        this.laRetenue=laRetenue;
        this.agenceId=agenceId;
    }
}

File: CondorV/CondorV/Models/BD/Site.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace CondorV.Models.BD
{
    public class Site
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; } 
        public string Nom { get; set; }
        public string LocalisationBarr { get; set; }
        public string Type { get; set; }
        public double Capacite { get; set; }
        public string VillePlusProche { get; set; }
        public double HauteurBarr { get; set; }
        public double DistVillePlusProche { get; set; }
        public string CodeRetNormal { get; set; }
        public DateTime DateMiseEnServ { get; set; }
        public string LaRetenue { get; set; }
        //Table HSV
        //But de Site
        [JsonIgnore]
        public ICollection<Utilisateur>? Utilisateurs { get; set; }
        public long? AgenceId { get; set; }
        //[JsonIgnore]
        public Agence? Agence { get; set; }
        [JsonIgnore]
        public ICollection<SitesGrandeurs>? SitesGrandeurs { get; set; }

        public Site() { }
    }
}
